manticore/core/smtlib/visitors.py

```diff
@@ -378,12 +378,15 @@ def visit_BitVecZeroExtend(self, expression, *operands):
             return expression
 
     def visit_BitVecITE(self, expression, *operands):
-        # FIXME Enable some taint propagating optimization
-        if isinstance(expression.operands[0], Constant) and not expression.operands[0].taint:
+        if isinstance(expression.operands[0], Constant):
             if expression.operands[0].value:
                 result = expression.operands[1]
             else:
                 result = expression.operands[2]
+            new_taint = result._taint | expression.operands[0].taint
```

Comment:
Why not `result = expressions.taint_with(result, expression.operands[0].taint)` ? Looks to be almost exactly the same code.

Reply:
I considered that, but:
1. I wasn't sure what exactly is going on in that function, e.g., this:
https://github.com/trailofbits/manticore/blob/19ba4b6339c350368cce904573caefb689a049f2/manticore/core/smtlib/expression.py#L95-L96
2. `taint_with` makes a copy of the expression every time.
I open to changing, though.